src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -1247,20 +1249,25 @@ BlankNode Visitor::visitTypesafe(Parser::BlankNodeContext* ctx) {
 }
 
 // ____________________________________________________________________________________
-template <typename Out, typename FirstContext, typename... Context>
-Out Visitor::visitAlternative(FirstContext ctx, Context... ctxs) {
+template <typename Out, bool isRecursive, typename FirstContext,
+          typename... Context>
+Out Visitor::visitAlternative(FirstContext* ctx, Context*... ctxs) {
+  if constexpr (!isRecursive) {
+    int validContexts =
+        (static_cast<bool>(ctx) + ... + static_cast<bool>(ctxs));
+    AD_CHECK(validContexts == 1);
+  }
   if (ctx) {
     return {visitTypesafe(ctx)};
   } else {
-    return visitAlternative<Out>(ctxs...);
+    if constexpr (sizeof...(Context) != 0) {
+      return visitAlternative<Out, true>(ctxs...);
+    } else {
+      AD_FAIL()  // Should be unreachable.
```

Comment:
I was going to suggest removing the if constexpr, but I found out (probably like you) that it is required for compilation... so maybe comment
`// unreachable because of the `AD_CHECK` above, but the `if constexpr` is still needed for the compilation of the base case`